homeassistant/components/aws/notify.py

```diff
@@ -0,0 +1,278 @@
+"""AWS platform for notify component."""
+import asyncio
+import logging
+import json
+import base64
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import CONF_PLATFORM, CONF_NAME, ATTR_CREDENTIALS
+from homeassistant.components.notify import (
+    ATTR_TARGET,
+    ATTR_TITLE,
+    ATTR_TITLE_DEFAULT,
+    BaseNotificationService,
+    PLATFORM_SCHEMA,
+)
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.json import JSONEncoder
+
+from .const import (
+    CONF_ACCESS_KEY_ID,
+    CONF_CREDENTIAL_NAME,
+    CONF_PROFILE_NAME,
+    CONF_REGION,
+    CONF_SECRET_ACCESS_KEY,
+    DATA_SESSIONS,
+)
+
+DEPENDENCIES = ["aws"]
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_CONTEXT = "context"
+CONF_SERVICE = "service"
+
+SUPPORTED_SERVICES = ["lambda", "sns", "sqs"]
+
+
+def _in_avilable_region(config):
+    """Check if region is available."""
+    import aiobotocore
+
+    session = aiobotocore.get_session()
+    available_regions = session.get_available_regions(config[CONF_SERVICE])
+    if config[CONF_REGION] not in available_regions:
+        raise vol.Invalid(
+            "Region {} is not available for {} service, mustin {}".format(
+                config[CONF_REGION], config[CONF_SERVICE], available_regions
+            )
+        )
+    return config
+
+
+PLATFORM_SCHEMA = vol.Schema(
+    vol.All(
+        PLATFORM_SCHEMA.extend(
+            {
+                # override notify.PLATFORM_SCHEMA.CONF_PLATFORM to Optional
+                # we don't need this field when we use discovery
+                vol.Optional(CONF_PLATFORM): cv.string,
+                vol.Required(CONF_SERVICE): vol.All(
+                    cv.string, vol.Lower, vol.In(SUPPORTED_SERVICES)
+                ),
+                vol.Required(CONF_REGION): vol.All(cv.string, vol.Lower),
+                vol.Inclusive(CONF_ACCESS_KEY_ID, ATTR_CREDENTIALS): cv.string,
+                vol.Inclusive(
+                    CONF_SECRET_ACCESS_KEY, ATTR_CREDENTIALS
+                ): cv.string,
+                vol.Exclusive(CONF_PROFILE_NAME, ATTR_CREDENTIALS): cv.string,
+                vol.Exclusive(
+                    CONF_CREDENTIAL_NAME, ATTR_CREDENTIALS
+                ): cv.string,
+                vol.Optional(CONF_CONTEXT): vol.Coerce(dict),
+            },
+            extra=vol.PREVENT_EXTRA,
+        ),
+        _in_avilable_region,
+    )
+)
+
+
+async def async_get_service(hass, config, discovery_info=None):
+    """Get the AWS notification service."""
+    import aiobotocore
+
+    session = None
+
+    if discovery_info is not None:
+        conf = discovery_info
+    else:
+        conf = config
```

Comment:
We should only allow set up via discovery from the aws component. So make this a guard clause instead, that checks if discovery_info is None and return None if so.